Bugs with rust 1.71.0-nightly

I updated my Rust toolchain:

```
$ cargo --version
cargo 1.71.0-nightly (de80432f0 2023-04-21)
```

Now I'm running into two problems (so far):

1. Feature `alloc_handler_error` no longer exists:

```
   Compiling linux_svsm v0.1.0 (/home/dov/linux-svsm)
error[E0557]: feature has been removed
  --> src/lib.rs:11:12
   |
11 | #![feature(alloc_error_handler)]
   |            ^^^^^^^^^^^^^^^^^^^ feature has been removed
   |
   = note: now handled by panic handler

error: cannot find attribute `alloc_error_handler` in this scope
    --> src/mem/alloc.rs:1145:3
     |
1145 | #[alloc_error_handler]
     |   ^^^^^^^^^^^^^^^^^^^
     |
     = note: `alloc_error_handler` is in scope, but it is a function, not an attribute

For more information about this error, try `rustc --explain E0557`.
error: could not compile `linux_svsm` (lib) due to 2 previous errors
```

It seems that the default implemenation of the alloc error is to panic, which is good for us (it'll call our custom panic handler). If I delete all mentions of alloc_error_handler, I can successfully build linux-svsm.

2. When I run linux-svsm, I get this:

```
Launching VM ...
  /tmp/cmdline.74246
char device redirected to /dev/pts/10 (label compat_monitor0)
> All 18 firmware config files read.
> Starting SMP for 7 APs:
-- AP 1/7 initialized.
-- AP 2/7 initialized.
-- AP 3/7 initialized.
-- AP 4/7 initialized.
-- AP 5/7 initialized.
-- AP 6/7 initialized.
-- AP 7/7 initialized.
PANIC!
panicked at 'misaligned pointer dereference: address must be a multiple of 0x4 but is 0xffff8000ffffff6e', src/bios.rs:203:23
PANIC!
```

I haven't yet started debugging this (or tried to understand why this behaviour has changed).

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -10,6 +10,7 @@ name = "linux_svsm"
 authors = ["Tom Lendacky <thomas.lendacky@amd.com>", "Carlos Bilbao <carlos.bilbao@amd.com>"]
 version = "0.1.0"
 edition = "2021"
+rust-version = "1.71"
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 

--- a/src/bios.rs
+++ b/src/bios.rs
@@ -13,6 +13,7 @@ use crate::*;
 
 use core::cmp::min;
 use core::mem::size_of;
+use core::ptr;
 use core::slice;
 use uuid::Bytes;
 use uuid::{uuid, Uuid};
@@ -200,7 +201,7 @@ fn find_bios_guid_entry(bios_info: &mut BiosInfo, target_guid: Uuid) -> Option<u
 }
 
 unsafe fn __find_snp_section(bios_info: &mut BiosInfo, stype: u32, p: u64) -> Option<SnpSection> {
-    let offset: u64 = *(p as *const u32) as u64;
+    let offset: u64 = ptr::read_unaligned(p as *const u32) as u64;
     if offset > bios_info.size() {
         return None;
     }

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -8,7 +8,6 @@
 
 #![feature(type_ascription)]
 #![feature(abi_x86_interrupt)]
-#![feature(alloc_error_handler)]
 // Disable the (implicitly-linked) standard library. #! defines behavior of the current module; as
 // we are in root, the entire crate is affected.
 #![no_std]

--- a/src/mem/alloc.rs
+++ b/src/mem/alloc.rs
@@ -1141,12 +1141,6 @@ unsafe impl GlobalAlloc for SvsmAllocator {
 #[cfg_attr(not(test), global_allocator)]
 pub static mut ALLOCATOR: SvsmAllocator = SvsmAllocator::new();
 
-#[cfg(not(test))]
-#[alloc_error_handler]
-fn alloc_error_handler(layout: alloc::alloc::Layout) -> ! {
-    panic!("Allocation failed: {:?}\n", layout)
-}
-
 /// Create a stack of size stack_pages and add a guard page
 /// also make the stack user accessible if needed
 pub fn mem_create_stack(stack_pages: u64, user_space: bool) -> VirtAddr {

